src/test/java/eu/europeana/metis/sandbox/common/TestUtils.java

```diff
@@ -25,4 +36,59 @@ public String readFileToString(String file) throws IOException {
     }
     return inputStream.readAllBytes();
   }
+
+  public List<ByteArrayInputStream> getContentFromZipFile(InputStream providedZipFile)
+      throws IOException {
+    String zipFolder = UUID.randomUUID().toString();
+    ZipFile zipFile = this.createTempZipFile(providedZipFile, zipFolder);
+
+    List<ByteArrayInputStream> result;
+    try {
+      List<InputStream> streams = this.getContentFromZipFile(zipFile);
+      List<ByteArrayInputStream> tmpResult = new ArrayList<>(streams.size());
+      Iterator<InputStream> streamsIterator = streams.iterator();
+
+      while (true) {
+        if (!streamsIterator.hasNext()) {
+          result = tmpResult;
+          break;
+        }
+        InputStream stream = streamsIterator.next();
+        tmpResult.add(new ByteArrayInputStream(IOUtils.toByteArray(stream)));
+      }
+    } catch (Throwable exceptionReading) {
+      try {
+        zipFile.close();
+        FileUtils.deleteDirectory(new File(zipFolder));
+      } catch (Throwable exceptionClosing) {
+        exceptionReading.addSuppressed(exceptionClosing);
+      }
+      throw exceptionReading;
+    }
+    return result;
+  }
+
+  private ZipFile createTempZipFile(InputStream content, String folder) throws IOException {
+    File tempFile = File.createTempFile(folder, ".zip");
+    FileUtils.copyInputStreamToFile(content, tempFile);
+    return new ZipFile(tempFile, 5);
+  }
+
+  private List<InputStream> getContentFromZipFile(ZipFile zipFile) throws IOException {
+    List<InputStream> result = new ArrayList<>();
+    Iterator<? extends ZipEntry> entries = zipFile.stream().iterator();
+
+    while (entries.hasNext()) {
```

Comment:
This `while` loop is a better implementation comparing to the previous one.
Why did you write this `.hasNext()` condition and the previous one as `while(true)`?

Reply:
The main reason for this was I didn't do/write this code. This was a copy/paste from previous ZipFileReader. I was more concerned with doing the right testing. And I agree with you, this code is messy. It was messier, I just removed all the esoteric variables names var1, var2, var3, etc and gave them more descriptive naming. I will follow you considerations

Reply:
I still see no changes in the loop, please do

Reply:
My bad, forgot to unstash changes